### PR TITLE
CI: catch malformed test cases

### PR DESCRIPTION
**Pull request by liushuyu**
- catch malformed test cases and use GitHub Actions' annotation feature to mark the failures

### CODE REVIEW

**Comment by liushuyu** [comment]
Demo:

Screenshot

**Comment by liushuyu** [comment]
> Does this script output some magic output to stdout that GitHub picks this up to annotate the files?

Yes. The script produces GitHub Actions Annotation Commands to the `stdout`. You can see the documentation for this "magic format" here: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions.

File: .github/emit_test_errors.pl [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    my $command = "::error file=gcc/testsuite/$filename";
+    if ($line) {
+        $command = "$command,line=$line";
+    }
+
+    print "$command,title=Test failure ($type)::$message\n";
+}
+
+analyze_errors();
```
Comment (CohenArthur) [inline .github/emit_test_errors.pl:25]: I honestly have no idea on how correct this script is and I personally cannot maintain it at all if something goes wrong at some point :DD But I'm fine with it if Philip is

**Review by CohenArthur — APPROVED** [review]

**Pull request merged**